view.py

```diff
@@ -1,20 +1,293 @@
+        combo_box.current(0)
+
+        #the back & next buttons
+        back_but = tk.Button(parent, text = "Back", bg = "Green", relief = "flat", 
+        command = lambda : main_view.show_frame(CreateUserDefinedMeal))
+        save_but = tk.Button(parent, text ="Save", bg = "Green", relief = "flat")
+        
+        # make all elements visible by using grid
+        parent.grid(column = 0, row = 0)
+        title.grid(column = 0, row = 0, columnspan = 3, pady = 10)
+        combo_box.grid (column = 1, row = 2, pady = 30)
+        back_but.grid(column = 0, row = 3, padx = 15, pady = 10)
+        save_but.grid(column = 2, row = 3, padx = 15, pady = 10)
+    
+    def save_database(self, main_view):
+        main_view.controller.save_user_info(main_view.user_info)
+        main_view.controller.save_user_defined_meal(main_view.user_defined_meal)
+        
+
+class CreateUserDefinedMeal(tk.Frame):
+
+    def __init__(self, parent, main_view):
+        tk.Frame.__init__(self, parent)
+        parent = tk.Frame(self, bg = "white")
+
+        container = ttk.Frame(parent)
+        canvas = tk.Canvas(container)
+        scrollbar = ttk.Scrollbar(container, orient="vertical", command=canvas.yview)
+        scrollable_frame = ttk.Frame(canvas)
+
+        scrollable_frame.bind(
+            "<Configure>",
+            lambda e: canvas.configure(
+                scrollregion=canvas.bbox("all")
+            )
+        )
+
+        canvas.create_window((0, 0), window=scrollable_frame, anchor="nw")
+
+        canvas.configure(yscrollcommand=scrollbar.set)
+        
+        # the title of the frame
+        title = tk.Label(parent,text = "Define your meals", font = TITLEFONT, bg = "white")
+        add_but = tk.Button(parent, text = "+", bg = "grey", command = lambda: self.add_sub_defined_meal(scrollable_frame, main_view))
+        back_but = tk.Button(parent, text = "Back", bg = "Green", relief = "flat",
+        command = lambda: main_view.show_frame(CreateBodyCondition))
+        next_but = tk.Button(parent, text ="Next", bg = "Green", relief = "flat",
+        command = lambda: main_view.show_frame(CreatePlanPeriod))
+        
+        
+        # make all elements visible by using grid
+        parent.grid(column = 0, row = 0)
+        title.grid(column = 0, row = 0, columnspan = 3, pady = 10)
+        add_but.grid(column = 2, row = 0)
+
+        container.grid(column = 1, row = 1)
+        canvas.pack(side="left", fill="both", expand=True)
+        scrollbar.pack(side="right", fill="y")
+
+
+        back_but.grid(column = 0, row = 3, padx = 15, pady = 10)
+        next_but.grid(column = 2, row = 3, padx = 15, pady = 10)
+
+
+    def add_sub_defined_meal(self, scrollable_frame, main_view):
+        new_meal = CreateSubUserDefinedMeal(self, scrollable_frame, main_view)
+        main_view.user_defined_meal.append(new_meal)
+        new_meal.pack() 
+
+class CreateSubUserDefinedMeal(tk.Frame):
+
+    def __init__(self, grandparent, parent, main_view):
```

Comment:
can you add a docstring to clarify the parent/grandparent of the function?

Reply:
Can this line run correctly? Because class CreateSubUserDefinedMeal uses this line of code to redefine the initial attribute from the inheritance class (which is CreateUserDefinedMeal class (parent) and MainView class (mainview)) . So if we don't change any factor in the grandparent, what is the point of a grandparent?